### ISSUE TITLE
False positive: dyn trait when referring Self`

### ISSUE THREAD

**Issue by 95th** [intellij-rust/intellij-rust#4230]
Intellij Rust show false positive about dyn keyword in traits:

```rust
pub trait TestFuture: Future {
    fn drive<T>(&mut self, other: T) -> Drive<'_, Self, T>
    where
        T: Future,
        Self: Future + Sized,
    {
        Drive {
            driver: self,
            future: Box::pin(other),
        }
    }
}
```

![Untitled](https://user-images.githubusercontent.com/22871609/62563011-b9eae280-b89f-11e9-9bcb-113f640fd4fa.png)

**Comment by Undin** [comment]
Duplicate of #4174
Fixed in #4175

**Comment by 95th** [comment]
@Undin Thanks. I updated my plugin but the issue is still happening. Is it released?

I am using IDEA community 2019.1

**Comment by Undin** [comment]
Not yet. It will be released next week

**Comment by Undin** [comment]
@95th Also, you can use nightly builds of the plugin. See [instruction](https://intellij-rust.github.io/install#nightly-builds) how to do it

**Comment by 95th** [comment]
@Undin Thanks!
